File: VB-Examples/21_Security/LockSpecificCellInNewExcel.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace LockSpecificCellInNewExcel
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Create an empty sheet in the Workbook
            workbook.CreateEmptySheet()

            ' Get the first worksheet from the Workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Loop through rows 0 to 254 and set the Locked property of each row's style to False
            For i As Integer = 0 To 254
                sheet.Rows(i).Style.Locked = False
            Next i

            ' Set the text of cell A1 in the worksheet to "Locked"
            sheet.Range("A1").Text = "Locked"

            ' Set the Locked property of cell A1's style to True
            sheet.Range("A1").Style.Locked = True

            ' Set the text of cells C1 to E3 in the worksheet to "Locked"
            sheet.Range("C1:E3").Text = "Locked"

            ' Set the Locked property of cells C1 to E3's style to True
            sheet.Range("C1:E3").Style.Locked = True

            ' Protect the worksheet with the specified password and enable all protection options
            sheet.Protect("123", SheetProtectionType.All)

            ' Specify the name of the resulting Excel file after locking specific cells
            Dim result As String = "Result-LockSpecificCellInNewlyXlsFile.xlsx"

            ' Save the Workbook to the specified path in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/21_Security/LockSpecificCellInNewExcel.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace LockSpecificCellInNewExcel
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook.
			Workbook workbook = new Workbook();

            // Create an empty worksheet.
            workbook.CreateEmptySheet();

            // Get the first worksheet.
			Worksheet sheet = workbook.Worksheets[0];

            // Loop through all the rows in the worksheet and unlock them.
            for (int i = 0; i < 255; i++)
            {
                sheet.Rows[i].Style.Locked = false;
            }

            // Lock specific cell in the worksheet.
            sheet.Range["A1"].Text = "Locked";
            sheet.Range["A1"].Style.Locked = true;

            // Lock specific cell range in the worksheet.
            sheet.Range["C1:E3"].Text = "Locked";
            sheet.Range["C1:E3"].Style.Locked = true;

            // Set the password.
            sheet.Protect("123", SheetProtectionType.All);

            // Specify the output filename for the workbook
            String result = "Result-LockSpecificCellInNewlyXlsFile.xlsx";

            // Save the modified workbook to a file (in Excel 2013 format)
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}
